### ISSUE TITLE
Add role icon property

### ISSUE THREAD

**Issue by quinchs** [Discord-Net-Labs/Discord.Net-Labs#154]
# Summary
Role icons are rolling out, Labs does not support them yet. Heres an example role
```jsonc
 {
        "id": "465089506127904768",
        "name": "Example",
        "permissions": "8052404161",
        "position": 163,
        "color": 3009399,
        "hoist": true,
        "managed": false,
        "mentionable": false,
        "icon": "8601b4b85075d23ccd20d79ca6223ba1", // new
        "unicode_emoji": null // new
    },
```

And the corresponding cdn path
```
https://cdn.discordapp.com/role-icons/{ROLE_ID}/{IMAGE_HASH}.png?size=32
```

**Comment by yutamago** [comment]
Is there any more information about this feature?

I added the basic properties and the CDN path, but I'm stuck on the image hash. Where would you get that?
And is there any way to test this feature right now?

I'm also wondering if the image format and size are freely settable.

**Comment by MrCakeSlayer** [comment]
You would get the hash (the icon property) from the role object like you would any of the other properties on the role. (Like how you would get the role color or if it's mentionable or not)

You can view the pending API documentation for it here: https://github.com/discord/discord-api-docs/pull/3815/files

You can also freely set the format and size as you would other Discord CDN images. An example being: https://cdn.discordapp.com/role-icons/465089506127904768/8601b4b85075d23ccd20d79ca6223ba1.jpg?size=128

**Comment by quinchs** [comment]
Closed in #204
